### PR TITLE
add fc/iscsi/nfs identifier tags to persistentvolume_info

### PR DESCRIPTION
**Pull request by jimmyseto**
**What this PR does / why we need it**:
export fc/iscsi/nfs info to help identify underlying storage

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

**Comment by tariq1890** [comment]
Could you provide documentation/link to where these fields are defined?

**Comment by jimmyseto** [comment]
> Could you provide documentation/link to where these fields are defined?

FC fields: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.18/#fcvolumesource-v1-core
ISCSI fields: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.18/#iscsipersistentvolumesource-v1-core
NFS fields: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.18/#nfsvolumesource-v1-core

File: internal/store/persistentvolume.go [hunk 1]
```diff
@@ -28,6 +28,8 @@ import (
 	"k8s.io/apimachinery/pkg/watch"
 	clientset "k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/tools/cache"
+
+	"strconv"
```
Comment (tariq1890) [inline internal/store/persistentvolume.go:32]: this needs to go under `"context"`

File: internal/store/persistentvolume_test.go [hunk 2]
```diff
@@ -169,7 +169,7 @@ func TestPersistentVolumeStore(t *testing.T) {
 			Want: `
 					# HELP kube_persistentvolume_info Information about persistentvolume.
 					# TYPE kube_persistentvolume_info gauge
-					kube_persistentvolume_info{ebs_volume_id="",gce_persistent_disk_name="",persistentvolume="test-pv-available",storageclass=""} 1
+					kube_persistentvolume_info{ebs_volume_id="",fc_lun="",fc_target_wwns="",fc_wwids="",gce_persistent_disk_name="",iscsi_iqn="",iscsi_lun="",iscsi_target_portal="",nfs_path="",nfs_server="",persistentvolume="test-pv-available",storageclass=""} 1
```
Comment (tariq1890) [inline internal/store/persistentvolume_test.go:172]: Can you add a test case, where atleast one of these label keys has a value? Just any one field would do

**Comment by tariq1890** [comment]
Thanks for this PR @jimmyseto 

Can you also make sure these fields are supported in `v1.14`. That's the minimum version we should be supporting

**Comment by jimmyseto** [comment]
> Thanks for this PR @jimmyseto
> 
> Can you also make sure these fields are supported in `v1.14`? That's the minimum version we should be supporting

Hi @tariq1890 -- Thanks for the quick response on this!

Confirmed that these fields are supported in v1.14.

FC fields: https://v1-14.docs.kubernetes.io/docs/reference/generated/kubernetes-api/v1.14/#fcvolumesource-v1-core
ISCSI fields: https://v1-14.docs.kubernetes.io/docs/reference/generated/kubernetes-api/v1.14/#iscsipersistentvolumesource-v1-core
NFS fields: https://v1-14.docs.kubernetes.io/docs/reference/generated/kubernetes-api/v1.14/#nfsvolumesource-v1-core

Also, updated the PR to address your CR comments/feedback above.

Thanks, again.

File: internal/store/persistentvolume.go (hunk 1 above)
Comment (tariq1890) [reply internal/store/persistentvolume.go:32]: This needs to be resolved @jimmyseto

Comment (jimmyseto) [reply internal/store/persistentvolume.go]: Sorry about that @tariq1890 .  The push was sitting at the login prompt.  The PR should be updated now.

File: internal/store/persistentvolume_test.go (hunk 2 above)
Comment (tariq1890) [reply internal/store/persistentvolume_test.go:172]: @jimmyseto You missed this one. Would be good to see a value populated for one of the new label keys added

**Comment by tariq1890** [comment]
Thanks very much @jimmyseto :)

**Comment by jimmyseto** [comment]
@tariq1890 - do you know if there is a target date for the next official release?

**Comment by lilic** [comment]
@jimmyseto  We have a release in progress, its a major breaking release so might take a few more days. https://github.com/kubernetes/kube-state-metrics/pull/1224

**Comment by jimmyseto** [comment]
> @jimmyseto We have a release in progress, its a major breaking release so might take a few more days. #1224

that's great, @lilic .  thanks for the link!

**Pull request merged**